### DISCUSSION TITLE
Looking for help (support, QA, maintainers)

### DISCUSSION THREAD

**Discussion by tony** [tmux-python/tmuxp#290] [General]
We have a lot of users between here and libtmux. We need to build a team.

Willing to pitch some personal time in to help make tmuxp better?

You're welcome to participate now! We're a well-documented, CI-backed open source project. You can start now by answering [issues](https://github.com/tmux-python/tmuxp/issues) and QA'ing / code reviewing [pull requests](https://github.com/tmux-python/tmuxp/pulls).

If you fit any of these characteristics, you're encouraged to reply here to become a helper / maintainer!

- love open source
- tmuxp user
- Python experience
- tmux experience
- can wrap their brain around the codebase (code quality between tmuxp and libtmux is good)
- Like of or interest in learning [pytest](https://docs.pytest.org/en/latest/)
- Like of or interest in documents (we use [sphinx](http://www.sphinx-doc.org/en/stable/))
- time they can devote to answering/closing issues
- QA'ing pull requests
- Assuring contributions maintain code standards ([flake8](http://flake8.pycqa.org/en/latest/), [vulture](https://github.com/jendrikseipp/vulture))
- Making sure tmuxp and libtmux stay true to what they're philosophically supposed to do, while allowing new features to come in.

Edit: mail me at tony at git dash pull dot com

**Comment by askedrelic** [comment]
I'm spending sometime today looking to fix #184 so maybe I can contribute. I use tmux/tmuxp daily and would like to do what I can do help things be active!

**Comment by tony** [comment]
Another call, need maintainers. 👍

**Comment by tony** [comment]
@askedrelic Are you still available/willing to help out around the project?

**Comment by askedrelic** [comment]
Hi @tony unfortunately I don't know how much time I can commit. I've been watching the tmuxp and libtmux projects the last few months, to get a better sense of things. I'll keep doing so and try to be more active on things.

Sidenote - the pyup and auto ticket mgmt bots are really nice for helping maintain the project!

**Comment by BlitzKraft** [comment]
I am interested to be a maintainer. I am not familiar with libtmux, yet. I have used and worked with tmuxp and would love to contribute.

**Comment by ritiek** [comment]
It probably won't make sense for me to be a maintainer since I am lack on time as well but I'll see around if I can help a bit whenever I get time and the mood. :D

**Comment by suprematis** [comment]
I can volunteer for performing QA.  Feel free to DM me.

**Comment by gunzapper** [comment]
I would help you. I could take a look at issues and also to the QA of merge request. Currently I use tmuxinator... but I am going to start to use tmuxp.

**Comment by tony** [comment]
@ritiek @suprematis @gunzapper If any of you are still interested please send me a mail. Anyone else who's interested see my email in the top post

**Comment by sloria** [comment]
@tony You might consider putting this project under the Jazzband org: https://jazzband.co/ .

**Reply by tony** [reply]
Jazzband is nice for some projects, I've contributed to a few. It's not ideal for ones where a lead developer or creator wants to maintain a final say.  With our independent organization https://github.com/tmux-python there is the best of both worlds, responsibility can be delegated out to maintainers while still having strong leadership. As an example we've had high level help QA'ing and publishing releases from our new maintainers - but I still release features on my own as an original creator / project lead (example: `tmuxp shell`) and adopt and improve contribution guidelines, can overrule decisions if its more efficient, etc.

**Comment by aRkedos** [comment]
@tony I was trying to get my feet wet and implemented #589, `flake8` checks out with no errors, but I can't finish `make test` see #620 :/

**Comment by tony** [comment]
@aRkedos thank you for the error report as well. Tests will run in CI but not sure you'd want to wait for that :)

**Comment by surfingdoggo** [comment]
@tony I can't commit regular time but I'm just (finally) starting to use `tmuxp` and as I get familiar, I'll start contributing!

**Reply by tony** [reply]
@surfingdoggo you are welcome!  We keep a [developer guide](https://tmuxp.git-pull.com/developing.html) open and are aiming to type annotate the codebase over the next few months. this should make catching mistakes easier

**Reply by tony** [reply]
P.S. as of 2 weeks ago, libtmux - the library tmuxp uses underneath the hood - has type annotations also

**Reply by surfingdoggo** [reply]
Awesome! I started reading through them already :D

**Comment by dagon666** [comment]
I'd be happy to help out.  Have some feature ideas already.  I've been trying to pitch https://github.com/tmux-python/tmuxp/discussions/1055 can't raise the PR though as I'm not in the maintainers group.

**Reply by tony** [reply]
when i originally wrote this, things were a lot different. The calculus has shifted a bit since then.

- i'm open to maintainers, but github / open source in general - and I too - are still normalizing to what defines help now
  - there may be a totally github org soon, tmuxp may or may not come along with it
- i will look into those and any issues

P.S. there's a large set of libtmux updates and tmuxp updates on the way
